### DISCUSSION TITLE
Filtrage d'entités dans la vue ecosystème

### DISCUSSION THREAD

**Discussion by jfcjm** [dbarzin/mercator#113] [Ideas]
Bonjour,
  la vue ecosystème peut être difficile à lire si le nombre d'entités est trop important => je suis en train de regarder s'il serait possible d'ajouter un menu permettant de filtrer les entités affichées  (menu similaire à celui utilisé dans la vue "information_system").

  J'envisage d'avoir les deux filtres  suivants  : 

1. entité interne ou externe à l'organisation étudiée
2. types d'entité.

La mise en oeuvre nécessite la modification Modele "Entite" :

1=> Ajout d'un attribut booléen Entity.is_external
2=> Ajout d'un attribut string Entity.type, les valeurs distinctes de ce champ sont affichées dans le menu de sélection correspondant.

 Ces valeurs sont ensuite utilisées pour filtrer les entités affichées; une relation sera alors représentée lorsque "à la fois" sa source et sa destination ne sont pas filtrées.

Votre avis ?

Jean-Marie
PS: j'ai actuellement le code pour le premier filtre (interne/externe) ...

**Comment by dbarzin** [comment]
Cela me semble être une bonne idée.
Vous maitrisez bien tout le framework Laravel pour faire ce changement ?
Il y a une migration à créer, ajouter les traductions, et mettre à jour les rapports.

**Reply by jfcjm** [reply]
Merc pour les conseils ...

Je ne m'y connais pas vraiment ni en php ni en laravel mais oui, après avoir jeté un coup d'oeil à la doc (et abusé de grep et de find :) , je pense être arrivé à cerner l'ensemble des modifs à faire (controllers, vues, cruds, rapports ...).  Je n'ai toutefois pas regardé l'aspect test unitaire/intégration (mais  le code semble tourner  sans pb).

Je peux peut-être définir une PR afin que  jetiez un coup d'oeil sur ce premier filtrage (entité interne/externe).

Jean-Marie

**Reply by dbarzin** [reply]
Il y a un très bon tutoriel sur Laravel publié par Laracast https://laracasts.com/series/laravel-8-from-scratch
